packages/react-native-sdk/src/components/Livestream/LivestreamControls/HostStartStreamButton.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  useCall,
+  useCallStateHooks,
+  useI18n,
+} from '@stream-io/video-react-bindings';
+import React, { useState } from 'react';
+import {
+  ActivityIndicator,
+  Pressable,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import { useTheme } from '../../../contexts';
+import { EndBroadcastIcon, StartStreamIcon } from '../../../icons';
+
+/**
+ * Props for the HostStartStreamButton component.
+ */
+export type HostStartStreamButtonProps = {
+  /**
+   * Handler to be called when the Start Stream button is pressed.
+   * @returns void
+   */
+  onStartStreamHandler?: () => void;
+  /**
+   * Handler to be called when the End Stream button is pressed.
+   * @returns void
+   */
+  onEndStreamHandler?: () => void;
+  /**
+   * Enable HTTP live streaming
+   */
+  hls?: boolean;
+};
+
+/**
+ * The HostStartStreamButton component displays and controls the start and end of the host's live stream.
+ */
+export const HostStartStreamButton = ({
+  onEndStreamHandler,
+  onStartStreamHandler,
+  hls,
+}: HostStartStreamButtonProps) => {
+  const [isAwaitingResponse, setIsAwaitingResponse] = useState(false);
+  const { useIsCallLive, useIsCallBroadcastingInProgress } =
+    useCallStateHooks();
+  const {
+    theme: {
+      colors,
+      variants: { iconSizes },
+      typefaces,
+      hostStartStreamButton,
+    },
+  } = useTheme();
+
+  const call = useCall();
+  const isCallLive = useIsCallLive();
+  const isCallBroadcasting = useIsCallBroadcastingInProgress();
+  const { t } = useI18n();
+
+  const liveOrBroadcasting = isCallLive || isCallBroadcasting;
+
+  const onStartStreamButtonPress = async () => {
+    if (onStartStreamHandler) {
+      onStartStreamHandler();
```

Comment:
shouldn't we return after this?

Reply:
All the other handler related props that we have in SDK, are implemented with purpose of being able to replace default logic. So I think we should be consistent.

Reply:
Good catch. Thanks for spotting it. 😄